src/session/content/message_row/media_picture.rs

```diff
@@ -0,0 +1,189 @@
+use gtk::{gdk, glib, graphene, prelude::*, subclass::prelude::*};
+
+const MIN_WIDTH: i32 = 100;
+const MIN_HEIGHT: i32 = 100;
+const MAX_HEIGHT: i32 = 400;
+
+mod imp {
+    use super::*;
+    use once_cell::sync::Lazy;
+    use std::cell::{Cell, RefCell};
+
+    #[derive(Debug, Default)]
+    pub struct MediaPicture {
```

Comment:
I find the name `Picture` inaccurate, if not wrong. It gives the impression that it is something static. However, this does not necessarily seem to be the case as PR #192 shows. Why not call it something like the thing it encapsulates - MediaPaintable?

Reply:
I know it's a bit of a weird naming, but it is consistent with the GtkPicture (the MediaPicture is basically that, but with a different sizing system). MediaPaintable would be confusional with GdkPaintable, which is a different thing.

Reply:
Oh, and I forgot to mention that GtkPicture is also used for videos (in fact it's used inside the GtkVideo widget), so that's why MediaPicture is basically the same thing.

Reply:
Yeah, I get your point now. But still so confusing :-P